tck/src/main/java/org/eclipse/microprofile/config/tck/ArrayConverterTest.java

```diff
@@ -1654,120 +1658,161 @@ public void testInstantSetInjection() {
-            new URL("http://openliberty.io"),
-            new URL("http://microprofile.io")});
+        assertURLArrayEquals(value);
         Optional<URL[]> optionalSingle = config.getOptionalValue("tck.config.test.javaconfig.converter.urlvalue", URL[].class);
         Assert.assertTrue(optionalSingle.isPresent());
         URL[] single = optionalSingle.get();
         Assert.assertNotNull(single);
         Assert.assertEquals(single.length, 1);
-        Assert.assertEquals(single, new URL[]{ new URL("http://microprofile.io") });
+        Assert.assertEquals(single[0].toURI(), new URL[]{ new URL("http://microprofile.io") }[0].toURI());
     }
 
     @Test
-    public void testUrlListLookupProgrammatically() throws MalformedURLException {
+    public void testUrlListLookupProgrammatically() throws MalformedURLException, URISyntaxException {
         List<URL> values = config.getValues("tck.config.test.javaconfig.converter.urlvalues", URL.class);
         Assert.assertNotNull(values);
         Assert.assertEquals(values.size(), 3);
-        Assert.assertEquals(values, Arrays.asList(
-            new URL("http://microprofile.io"),
-            new URL("http://openliberty.io"),
-            new URL("http://microprofile.io")));
+        assertURLListEquals( values);
         List<URL> single = config.getValues("tck.config.test.javaconfig.converter.urlvalue", URL.class);
         Assert.assertNotNull(single);
         Assert.assertEquals(single.size(), 1);
-        Assert.assertEquals(single, Arrays.asList(new URL("http://microprofile.io")));
+        Assert.assertEquals(single.get(0).toURI(), Arrays.asList(new URL("http://microprofile.io")).get(0).toURI());
     }
 
     @Test
-    public void testOptionalUrlListLookupProgrammatically() throws MalformedURLException {
+    public void testOptionalUrlListLookupProgrammatically() throws MalformedURLException, URISyntaxException {
         Optional<List<URL>> optionalValues = config.getOptionalValues("tck.config.test.javaconfig.converter.urlvalues", URL.class);
         Assert.assertTrue(optionalValues.isPresent());
         List<URL> values = optionalValues.get();
         Assert.assertNotNull(values);
         Assert.assertEquals(values.size(), 3);
-        Assert.assertEquals(values, Arrays.asList(
-            new URL("http://microprofile.io"),
-            new URL("http://openliberty.io"),
-            new URL("http://microprofile.io")));
+        assertURLListEquals(values);
         Optional<List<URL>> optionalSingle = config.getOptionalValues("tck.config.test.javaconfig.converter.urlvalue", URL.class);
         Assert.assertTrue(optionalSingle.isPresent());
         List<URL> single = optionalSingle.get();
         Assert.assertNotNull(single);
         Assert.assertEquals(single.size(), 1);
-        Assert.assertEquals(single, Arrays.asList(new URL("http://microprofile.io")));
+        Assert.assertEquals(single.get(0).toURI(), Arrays.asList(new URL("http://microprofile.io")).get(0).toURI());
```

Comment:
`Arrays.asList(new URL("http://microprofile.io")).get(0)` should just be `new URL("http://microprofile.io")`